File: fullstack-front-end/src/app/contract/ticket.ts
import { User } from "./User";

export interface Ticket {
  id: number | null;
  subject: string;             // Required, max 100 chars
  description: string;         // Required, max 1000 chars
  priority: string;    // Required (enum)
  createdBy: number | null;           // Required (user ID)
  assignedTo?: number | null;  // Optional or null (user ID)
  createdAt: Date;             // Java LocalDate → TypeScript Date
  status: string;        // Optional or enum
}

File: Customer-Support-Ticket-Management-System-1/src/main/java/com/dollop/app/model/Ticket.java
package com.dollop.app.model;

import java.time.LocalDate;

import com.fasterxml.jackson.annotation.JsonFormat;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
  
@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Ticket {
	
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;

	private String subject;

	private String description;

	
	private TicketPriority priority;

	@ManyToOne
	private Users createdBy;

	@ManyToOne
	private Users assignedTo;

	@Temporal(TemporalType.DATE)
	@JsonFormat(pattern = "dd-MM-yyyy")
	private LocalDate createdAt = LocalDate.now();
	
	private TicketStatus status;
}

	
	


